java/ql/lib/semmle/code/java/dataflow/ExternalFlow.qll

```diff
@@ -283,7 +327,11 @@ module ModelValidation {
         ] and
       not kind.matches("regex-use%") and
       not kind.matches("qltest%") and
-      result = "Invalid kind \"" + kind + "\" in sink model."
+      msg = "Invalid kind \"" + kind + "\" in sink model." and
+      // The deprecation part of this message can be deleted after June 1st, 2024.
+      if kind instanceof DeprecatedSinkKind
+      then result = msg + " " + kind.(DeprecatedSinkKind).deprecationMessage()
```

Comment:
Does this mean it's an error to use a deprecated sink kind? Usually, deprecation means something is still available, but it is scheduled to be removed in the future. I don't think this is happening here. If my understanding is correct. maybe it's better to avoid the use of the term "Deprecated" and use something else like removed or out-dated?

Reply:
Yes, good point, I've updated the wording to use "outdated" in https://github.com/github/codeql/pull/13228/commits/b8cedfa817b6b4bc8d56cad8d3cf6faf4da69b01.

Reply:
Yes, good point, I've updated the wording to use "outdated" in https://github.com/github/codeql/pull/13228/commits/b8cedfa817b6b4bc8d56cad8d3cf6faf4da69b01.